scipy/signal/spectral.py

```diff
@@ -766,6 +771,18 @@ def _spectral_helper(x, y, fs=1.0, window='hann', nperseg=256,
                 pad_shape[-1] = x.shape[-1] - y.shape[-1]
                 y = np.concatenate((y, np.zeros(pad_shape)), -1)
 
+    #parse window; if array like, then set nperseg = win.shape
+    if isinstance(window, string_types) or type(window) is tuple:
```

Comment:
same comments about `ininstance` and `nperseg is None`.

Any way these checks could be refactored into a private function instead of repeated multiple places?